### DISCUSSION TITLE
All footnotes on one page

### DISCUSSION THREAD

**Discussion by yochem** [typst/typst#2974] [Q&A] [answered]
Is there a way to have a footnote page at the end of the document? I tried:

1. using `query`, but footnote didn't contain the footnote numbers and `footnote.entry` is not locatable
2. using `outline`, but `footnote` can't be a target and `footnote.entry` is not locatable

**Comment by yochem** [comment]
Current hack is just asssuming the footnotes are in order and enumerating the `query` output.

**Comment by YDX-2147483647** [comment, marked as answer]
I search Discord and find…

- [Footnotes & Endnotes, qjcg/awesome-typst](https://github.com/qjcg/awesome-typst#footnotes--endnotes)

  - [tudborg/notes.typ: Footnotes + Endnotes + Custom notes for Typst typesetting.](https://github.com/tudborg/notes.typ)
  - [saadulkh/typst-notes: Provides footnotes/endnotes implementation in Typst](https://github.com/saadulkh/typst-notes) (Archived on Jun 7, 2023)

- [Dedicated Notes Function(s) · typst/typst · Discussion #1335](https://github.com/typst/typst/discussions/1335)

  > Footnotes were added in v0.4.0 (`#footnote[...]`). The rest are probably possible to implement in libraries, but make for interesting ideas. Perhaps open a feature request issue for them?

Besides, if you need bibliography, you can use [`#bibliography("○○.bib")`](https://typst.app//docs/reference/model/bibliography).
